drivers/gpio/gpio_rt1718s_port.c

```diff
@@ -0,0 +1,372 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT richtek_rt1718s_gpio_port
+
+/**
+ * @file Driver for RS1718S TCPC chip GPIOs.
+ */
+
+#include "gpio_utils.h"
+
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/gpio/rt1718s.h>
+#include <zephyr/logging/log.h>
+LOG_MODULE_DECLARE(gpio_rt1718s_port, CONFIG_GPIO_LOG_LEVEL);
+
+/* Driver config */
+struct gpio_rt1718s_port_config {
+	/* gpio_driver_config needs to be first */
+	struct gpio_driver_config common;
+	/* RT1718S chip device */
+	const struct device *rt1718s_dev;
+};
+
+/* Driver data */
+struct gpio_rt1718s_port_data {
+	/* gpio_driver_data needs to be first */
+	struct gpio_driver_data common;
+	/* GPIO callback list */
+	sys_slist_t cb_list_gpio;
+	/* lock GPIO registers access */
+	struct k_sem lock;
+};
+
+/* GPIO api functions */
+static int gpio_rt1718s_pin_config(const struct device *dev, gpio_pin_t pin, gpio_flags_t flags)
+{
+	const struct gpio_rt1718s_port_config *const config = dev->config;
+	struct gpio_rt1718s_port_data *const data = dev->data;
+	uint8_t new_reg = 0;
+	int ret = 0;
+
+	/* Don't support simultaneous in/out mode */
+	if (((flags & GPIO_INPUT) != 0) && ((flags & GPIO_OUTPUT) != 0)) {
```

Comment:
The `!= 0` should be redundant here right? You could just do `if ((flags & GPIO_INPUT) && (flags & GPIO_OUTPUT))` or even better `if (flags & (GPIO_INPUT | GPIO_OUTPUT))`. Same below.